### PR TITLE
feat: add support for a number of exposure-related options

### PR DESCRIPTION
**Pull request by darrylhodgins**
This is related to #8

### CODE REVIEW

File: src/lib/shared-args.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * Auto White Balance Mode
+     */
+    ...(options.awbMode ? ['--awb', options.awbMode.toString()] : []),
+
+    /**
+     * Analog Gain
+     */
+    ...(options.analoggain ? ['--analoggain', options.analoggain.toString()] : []),
```
Comment (gregnr) [inline src/lib/shared-args.ts:74]: Let's stay consistent with this lib's camel case convention (even though the underlying arg isn't camel cased)
```suggestion
    ...(options.analogGain ? ['--analoggain', options.analogGain.toString()] : []),
```

File: src/lib/shared-args.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * Digital Gain
+     */
+    ...(options.digitalgain ? ['--digitalgain', options.digitalgain.toString()] : [])
```
Comment (gregnr) [inline src/lib/shared-args.ts:79]:
```suggestion
    ...(options.digitalGain ? ['--digitalgain', options.digitalGain.toString()] : [])
```

File: src/lib/still-camera.ts [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
   rotation?: Rotation;
   flip?: Flip;
   delay?: number;
+  shutter?: number;
+  iso?: number;
+  exposureCompensation?: number;
+  exposureMode?: ExposureMode;
+  awbMode?: AwbMode;
+  analoggain?: number;
+  digitalgain?: number;
```
Comment (gregnr) [inline src/lib/still-camera.ts:18]:
```suggestion
  analogGain?: number;
  digitalGain?: number;
```

File: src/lib/stream-camera.ts [hunk 4]
```diff
@@ -29,6 +30,13 @@ export interface StreamOptions {
   fps?: number;
   codec?: Codec;
   sensorMode?: SensorMode;
+  shutter?: number;
+  iso?: number;
+  exposureCompensation?: number;
+  exposureMode?: ExposureMode;
+  awbMode?: AwbMode;
+  analoggain?: number;
+  digitalgain?: number;
```
Comment (gregnr) [inline src/lib/stream-camera.ts:39]:
```suggestion
  analogGain?: number;
  digitalGain?: number;
```

File: README.md [hunk 5]
```diff
@@ -239,6 +239,13 @@ const stillCamera = new StillCamera({
 - [`rotation: Rotation`](#rotation) - *Default: `Rotation.Rotate0`*
 - [`flip: Flip`](#flip) - *Default: `Flip.None`*
 - `delay: number` - *Default: 1 ms*
+- `shutter: number` - *Default: Auto calculated based on framerate (1000000µs/fps). Number is in microseconds*
+- `iso: number` - *Default: Auto*
+- `exposureCompensation: number` - *Default: `0`*
+- [`exposureMode: ExposureMode`](#exposuremode) - *Default: Auto*
+- [`awbMode: AwbMode`](#awbmode) - *Default: Auto*
+- `analoggain: number` - *Default: 0*
+- `digitalgain: number` - *Default: 0*
```
Comment (gregnr) [inline README.md:248]:
```suggestion
- `analogGain: number` - *Default: 0*
- `digitalGain: number` - *Default: 0*
```

File: README.md [hunk 6]
```diff
@@ -271,6 +278,13 @@ const streamCamera = new StreamCamera({
 - `fps: number` - *Default: 30 fps*
 - [`codec: Codec`](#codec) - *Default: `Codec.H264`*
 - [`sensorMode: SensorMode`](#sensormode) - *Default: `SensorMode.AutoSelect`*
+- `shutter: number` - *Default: Auto calculated based on framerate (1000000µs/fps). Number is in microseconds*
+- `iso: number` - *Default: Auto*
+- `exposureCompensation: number` - *Default: `0`*
+- [`exposureMode: ExposureMode`](#exposuremode) - *Default: Auto*
+- [`awbMode: AwbMode`](#awbmode) - *Default: Auto*
+- `analoggain: number` - *Default: 0*
+- `digitalgain: number` - *Default: 0*
```
Comment (gregnr) [inline README.md:287]:
```suggestion
- `analogGain: number` - *Default: 0*
- `digitalGain: number` - *Default: 0*
```

File: README.md [hunk 7]
```diff
@@ -345,6 +359,44 @@ Image flip options.
...
+- `Auto`
+- `Sun`
+- `Cloud`
+- `Shade`
+- `Tungsten`
+- `Fluorescent`
+- `Incandescent`
+- `Flash`
+- `Horizon`
+- `GreyWorld`
```
Comment (gregnr) [inline README.md:394]:
```suggestion
- `AwbMode.Off`
- `AwbMode.Auto`
- `AwbMode.Sun`
- `AwbMode.Cloud`
- `AwbMode.Shade`
- `AwbMode.Tungsten`
- `AwbMode.Fluorescent`
- `AwbMode.Incandescent`
- `AwbMode.Flash`
- `AwbMode.Horizon`
- `AwbMode.GreyWorld`
```

File: README.md [hunk 8]
```diff
@@ -345,6 +359,44 @@ Image flip options.
 import { Flip } from "pi-camera-connect";
 ```
 
+## `ExposureMode`
+Exposure mode options.
+- `ExposureMode.Off`
```
Comment (gregnr) [inline README.md:364]: I don't see `off` as an exposure mode option in the docs. Can you please confirm that this exists?

File: src/lib/shared-args.ts (hunk 1 above)
Comment (darrylhodgins) [reply src/lib/shared-args.ts:74]: Oops, that was a copy-paste/multi-line edit thing.  I'll fix that.

File: README.md (hunk 8 above)
Comment (darrylhodgins) [reply README.md:364]: It's in the CLI help, and I am using this option for my usecase, and it seems to work…

```bash
$ raspistill --help

--snipped--

-ex, --exposure	: Set exposure mode (see Notes)

--snipped--

Notes

Exposure mode options :
off,auto,night,nightpreview,backlight,spotlight,sports,snow,beach,verylong,fixedfps,antishake,fireworks

--snipped--
```

I've never had success using the `off` option for _white balance_, though (`--awb`)… everything looks green when I do that.

Comment (gregnr) [reply README.md:364]: Okay thanks for confirming. Looks like their online docs are a bit out of sync...

https://www.raspberrypi.org/documentation/raspbian/applications/camera.md

I'll make a note to reference the CLI going forward.

File: README.md [hunk 9]
```diff
@@ -345,6 +361,44 @@ Image flip options.
...
+- `Awb.Auto`
+- `Awb.Sun`
+- `Awb.Cloud`
+- `Awb.Shade`
+- `Awb.Tungsten`
+- `Awb.Fluorescent`
+- `Awb.Incandescent`
+- `Awb.Flash`
+- `Awb.Horizon`
+- `Awb.GreyWorld`
```
Comment (gregnr) [inline README.md:396]:
```suggestion
- `AwbMode.Off`
- `AwbMode.Auto`
- `AwbMode.Sun`
- `AwbMode.Cloud`
- `AwbMode.Shade`
- `AwbMode.Tungsten`
- `AwbMode.Fluorescent`
- `AwbMode.Incandescent`
- `AwbMode.Flash`
- `AwbMode.Horizon`
- `AwbMode.GreyWorld`
```

Comment (darrylhodgins) [reply README.md]: D'oh!  Thanks for catching that :)

**Comment by gregnr** [comment]
Great work. Thanks again!

**Comment by darrylhodgins** [comment]
You're very welcome… I needed a few of these options for my project, so it helps me out too :)

**Pull request merged**